### PR TITLE
Fix dropdown keys to open menu

### LINKED ISSUE #32714: Dropdown up/down keys no longer open menu
Noticed that when the dropdown button toggle is focused, pressing the "up" or "down" keys no longer open the dropdown menu.  In v4, this used to work.  I'm unsure if this was an intentional change - I didn't see any mention of this in the docs.

This can be observed in the basic dropdown button example in docs:
https://getbootstrap.com/docs/5.0/components/dropdowns/#single-button

Here's the v4 example for reference:
https://getbootstrap.com/docs/4.5/components/dropdowns/#single-button

- Operating system and version: Windows
- Browser and version: All

### PR DESCRIPTION
**Pull request by sijusamson**
Fixes #32714.

### CODE REVIEW

**Comment by XhmikosR** [comment]
Can you add a test please? This wasn't caught.

**Comment by sijusamson** [comment]
> Can you add a test please? This wasn't caught.

Done !

**Comment by XhmikosR** [comment]
Works good, thanks!

@patrickhlauke anything a11y-wise we might be missing?
@rohit2sharma95: LGTY?

**Comment by patrickhlauke** [comment]
It's more a nice to have than a necessary (from a11y point of view) feature, but conversely it's not introducing any issues either. Long way of saying LGTM.

File: js/tests/unit/dropdown.spec.js [hunk 1]
```diff
@@ -1582,4 +1582,31 @@ describe('Dropdown', () => {
...
+      '    <a id="item1" class="dropdown-item" href="#">Another link</a>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const triggerDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+    const dropdown = fixtureEl.querySelector('.dropdown')
+
+    const keydown = createEvent('keydown')
+    keydown.key = 'ArrowDown'
```
Comment (rohit2sharma95) [inline js/tests/unit/dropdown.spec.js:1602]: It only checks that dropdown is shown if ArrowDown key is pressed, better if you add a check for ArrowUp as well :slightly_smiling_face:

File: js/src/dropdown.js [hunk 2]
```diff
@@ -450,6 +450,12 @@ class Dropdown extends BaseComponent {
       return
     }
 
+    if (!isActive && (event.key === ARROW_UP_KEY || event.key === ARROW_DOWN_KEY)) {
+      const button = this.matches(SELECTOR_DATA_TOGGLE) ? this : SelectorEngine.prev(this, SELECTOR_DATA_TOGGLE)[0]
+      button.click()
+      return
+    }
+
```
Comment (rohit2sharma95) [inline js/src/dropdown.js:458]: Now the method `dataApiKeydownHandler` has a linting warning:
_Static method 'dataApiKeydownHandler' has a complexity of 23. Maximum allowed is 20._

**Review by rohit2sharma95 — APPROVED** [review]

File: js/src/dropdown.js (hunk 2 above)
Comment (sijusamson) [reply js/src/dropdown.js:458]: @rohit2sharma95 Thank you for your suggestion, let me try to figure out the best possible solution.

Comment (sijusamson) [reply js/src/dropdown.js:458]: As per my findings, this linting warning can be removed if we split the function `dataApiKeydownHandler`. We can create a new static method, and pass on some functionality there. 
But should we do that ?
@rohit2sharma95 @patrickhlauke @XhmikosR

Comment (XhmikosR) [reply js/src/dropdown.js:478]: I'd say ignore it for now and tackle it later in another PR.

**Comment by XhmikosR** [comment]
BTW does it make sense to open the dropdown but not select the first/last element depending on the arrow key?

I feel like it's an extra key. But that's for later too. :)

**Pull request merged**